src/Symfony/Component/Yaml/Tests/InlineTest.php

```diff
@@ -420,4 +420,14 @@ public function testVeryLongQuotedStrings()
 
         $this->assertEquals($longStringWithQuotes, $arrayFromYaml['longStringWithQuotes']);
     }
+
+    public function testBooleanMappingKeysAreConvertedToStrings()
+    {
+        $this->assertSame(array('false' => 'foo'), Inline::parse('{false: foo}'));
```

Comment:
@GuilhemN our current behavior is to avoid evaluating keys (as PHP does not support arbitrary keys). Changing this to throw exceptions for any non integer or string keys is a big BC break, and I'm not sure this one would actually benefit users that much.